lib/chef/resource/aws_internet_gateway.rb

```diff
@@ -42,7 +42,9 @@ class Chef::Resource::AwsInternetGateway < Chef::Provisioning::AWSDriver::AWSRes
 
   def aws_object
     driver, id = get_driver_and_id
-    result = driver.ec2.internet_gateways[id] if id
-    result && result.exists? ? result : nil
+    ec2_resource = ::Aws::EC2::Resource.new(driver.ec2)
+    result = ec2_resource.internet_gateway(id) if id
+    result && result.data ? result : nil
```

Comment:
The exception handling should be done for this block which was removed in the previous commit.